### PR TITLE
Add plant relations to scraper

### PR DESCRIPTION
**Pull request by temmey**
## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [x] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation
- [x] I fixed all affected decisions
- [ ] I added unit tests for my code
- [x] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [x] Code is conforming to [our Architecture](/doc/architecture)
- [x] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [x] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: scraper/README.md [hunk 1]
```diff
@@ -100,5 +100,16 @@ The corrected data in the new file should be stored in the same format as the ge
 The scraper also inserts the scraped data into the database:
 
 ```shell
-npm run insert
```
Comment (markus2330) [inline scraper/README.md:103]: A `npm run insert` which does both insert:plants and insert:relations would be handy, too (and documented). Ideally the `npm run insert` in the scraper should be also added to Makefile.
ci/Jenkinsfile line 190 must be adapted to also copy the Companions.csv` and `Antigonist.csv` files.

File: scraper/src/insert_plant_relations.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+const db = pgp(process.env.DATABASE_URL);
+const companionsFilePath = process.argv[2] || 'data/Companions.csv';
+const antagonistFilePath = process.argv[3] || 'data/Antagonist.csv';
+
+/**
+ * Get Ids via the unique plant names in the database and populate a map 
+ *
+ * @param {*} map - a map with all unique plantnames and empty values
+ * 
+ * todo for later: user promise.all to requests all ids at the same time, and not wait for them
```
Comment (markus2330) [inline scraper/src/insert_plant_relations.js:17]: How is the current performance?

File: scraper/src/insert_plant_relations.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    return {
+    'plant1': map.get(item["Plant 1 (unique name)"]),
+    'plant2': map.get(item["Plant 2 (unique name)"]),
+    'relation': relation,
+    'note': "Imported via scrapper"
+    }
+  });
+
+  /**
+   * there are cases when we have multiple rows with the same relation, but diffrent common names. eg:
```
Comment (markus2330) [inline scraper/src/insert_plant_relations.js:60]:
```suggestion
   * there are cases when we have multiple rows with the same relation, but different common names. e.g:
```

File: scraper/src/insert_plant_relations.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+   * "Aubergine, Eggplant",Solanum melongena,Coriandrum sativum,Coriander,,,,,
+   * "Aubergine, Eggplant",Solanum melongena,Coriandrum sativum,Cilantro,,,,,
+   * thats why i added the ON CONFLICT DO NOTHING Clause.
```
Comment (markus2330) [inline scraper/src/insert_plant_relations.js:63]: Is this consistent with the other scraper's behavior? What happens on reimporting changed files? Please document in README.md

**Review by markus2330 — APPROVED** [review]
Great job, CI integration is missing but this can be done in follow up. If wanted that way, please create follow-up issue.

File: scraper/src/insert_plant_relations.js (hunk 2 above)
Comment (temmey) [reply scraper/src/insert_plant_relations.js:17]: For the current dataset, this part took 118.88ms when I timed it. It's just a part where I can see theoretical improvement.

Comment (markus2330) [reply scraper/src/insert_plant_relations.js:17]: hehe, so I don't think we need that todo for now :wink:

File: scraper/src/insert_plant_relations.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * Insert Plant Relations into database.
+ *
+ * @param {*} jsonArray - an json array with all the csv informations
+ * @param {*} map - a map with all unique plantnames and empty values
+ * @param {*} relation - the relation we are adding atm. can be companion, neutral or antagonist. Its a ENUM in our database
+ *
+ */
+
```
Comment (badnames) [inline scraper/src/insert_plant_relations.js:36]:
```suggestion
```

**Review by badnames — APPROVED** [review]

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
@4ydan this is the PR (and #645 is the issue) we talked about today. CI basically only needs to copy two files (or three to be future-proof for the next extension), would be great if we find a solution soon.

**Comment by 4ydan** [comment]
Currently testing the [solution](https://build.libelektra.org/job/PermaPlanT-Folder/job/PermaplanT/job/feature%2F645-ci-plant-relations/1/display/redirect). Will provide you with the lines for the Jenkinsfile, once it passes.

**Comment by 4ydan** [comment]
It seems to work now. https://build.libelektra.org/job/PermaplanT/job/feature%252F645-ci-plant-relations/

Replace this:
https://github.com/ElektraInitiative/PermaplanT/blob/90c043f241ecf8dbdc3204e149a8f01cfa3429f4/ci/Jenkinsfile#L201-L208

With this:
```suggestion
dir('scraper') {
     sh 'npm ci'
     sh 'mkdir ./data/'
     sh 'cp /nextcloud/mergedDatasets.csv ./data/'
     sh 'cp /nextcloud/Companions.csv ./data/'
     sh 'cp /nextcloud/Antagonist.csv ./data/'
     // Create empty file Sizes.csv for now, replace later with cp
     sh 'touch ./data/Sizes.csv'
     sh 'npm run insert'
     sh 'rm -rf ./data/'
     sh 'rm -rf ./node_modules/'
 }
```

**Comment by temmey** [comment]
When attempting to retrieve details about the build failure, I encounter 404 errors thrown by Jenkins.

**Comment by 4ydan** [comment]
Thank you, we moved the pipelines and I didn't check the anon settings. Should work now again. Sorry!

**Comment by 4ydan** [comment]
> @4ydan you can also push to PRs,

Oki
Pipeline seems to pass now 👍🏻

**Comment by markus2330** [comment]
Great! :rocket:

**Pull request merged**